crypto/pkcs7/pk7_smime.c

```diff
@@ -322,7 +322,7 @@ int PKCS7_verify(PKCS7 *p7, STACK_OF(X509) *certs, X509_STORE *store,
         tmpout = out;
 
     /* We now have to 'read' from p7bio to calculate digests etc. */
-    if ((buf = OPENSSL_malloc(BUFFERSIZE)) == NULL) {
+    if ((buf = OPENSSL_zalloc(BUFFERSIZE)) == NULL) {
```

Comment:
not sure I understand why this needs to be done here

Reply:
The Coverity issue was "String not null terminated". Looking at the details more closely...  the bio callbacks used for read and write ends up using strlen which expects the null terminated buffer. 
I think we can ignore this one and have it fixed in the callback.